### ISSUE TITLE
Unable to run tests after creating a maven Gauge project

### ISSUE THREAD

**Issue by kashishm** [getgauge/gauge#563]
_From @sguptatw on February 13, 2017 3:39_

**Expected behavior**
Should be able to run tests after creating a Gauge project

**Actual behavior**
Getting an exception

**Steps to replicate**
1. In Windows create a gauge project using command
`gauge --init java`
2. run the gauge tests using command
`mvn clean test`
3. Getting the following exception

```
[INFO]
[INFO] --- gauge-maven-plugin:1.1.0:execute (default) @ art ---
[ValidationError] C:\work\testGaugeTemplates\java\art\specs\example.spec:11 Step implementation not found => 'Vowels in English language are "aeiou".'
[ValidationError] C:\work\testGaugeTemplates\java\art\specs\example.spec:18 Step implementation not found => 'The word "gauge" has "3" vowels.'
[ValidationError] C:\work\testGaugeTemplates\java\art\specs\example.spec:28 Step implementation not found => 'Almost all words have vowels <table>'
Unable to start plugin html-report. Plugin html-report is not installed. To install, run `gauge --install html-report`.
com.google.protobuf.InvalidProtocolBufferException: Message missing required fields: messageType
        at com.google.protobuf.UninitializedMessageException.asInvalidProtocolBufferException(UninitializedMessageException.java:81)
        at com.google.protobuf.AbstractParser.checkMessageInitialized(AbstractParser.java:71)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:168)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:180)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:185)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:49)
        at gauge.messages.Messages$Message.parseFrom(Messages.java:21442)
        at com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:67)
        at com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:37)
com.google.protobuf.InvalidProtocolBufferException: Message missing required fields: messageType

  Error Message: Connection closed [127.0.0.1:55455] cause: read tcp 127.0.0.1:55451->127.0.0.1:55455: wsarecv: An existing connection was forcibly closed by the remote host.
  Stacktrace:


  Error Message: write tcp 127.0.0.1:55451->127.0.0.1:55455: use of closed network connection
  Stacktrace:

Specifications: 0 executed      0 passed        0 failed        0 skipped
Scenarios:      0 executed      0 passed        0 failed        0 skipped

Total time taken: 109ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 56.977 s
[INFO] Finished at: 2017-02-13T09:05:19+05:30
[INFO] Final Memory: 19M/248M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.thoughtworks.gauge.maven:gauge-maven-plugin:1.1.0:execute (default) on project art: Gauge Specs execution failed -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```
**Version**
```
Gauge version: 0.8.0.nightly-2017-02-09

Plugins
-------
csharp (0.9.3.nightly-2017-01-30)
java (0.6.0.nightly-2017-02-11)
```

_Copied from original issue: getgauge/gauge-csharp#107_

**Comment by kashishm** [comment]
_From @sguptatw on February 13, 2017 5:2_

Getting the following exception after updating the Gauge maven java.
```
  Error Message: java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://github.com/SeleniumHQ/
selenium/wiki/ChromeDriver. The latest version can be downloaded from http://chromedriver.storage.googleapis.com/index.html
  Stacktrace:
  com.google.common.base.Preconditions.checkState(Preconditions.java:177)
  org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:109)
  org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:32)
  org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:137)
  org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:296)
  org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:88)
  org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:116)
  driver.DriverFactory.getDriver(DriverFactory.java:24)
  driver.Driver.initializeDriver(Driver.java:16)
  sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  java.lang.reflect.Method.invoke(Method.java:498)
  com.thoughtworks.gauge.execution.MethodExecutor.execute(MethodExecutor.java:32)
  com.thoughtworks.gauge.execution.HooksExecutor$TaggedHookExecutor.executeHook(HooksExecutor.java:99)
  com.thoughtworks.gauge.execution.HooksExecutor$TaggedHookExecutor.execute(HooksExecutor.java:85)
  com.thoughtworks.gauge.execution.HooksExecutor.execute(HooksExecutor.java:42)
  com.thoughtworks.gauge.processor.MethodExecutionMessageProcessor.executeHooks(MethodExecutionMessageProcessor.java:54)
  com.thoughtworks.gauge.processor.SuiteExecutionStartingProcessor.process(SuiteExecutionStartingProcessor.java:26)
  com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:86)
  com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:37)


  Error Message: java.lang.NullPointerException
  Stacktrace:
  driver.Driver.closeDriver(Driver.java:22)
  sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  java.lang.reflect.Method.invoke(Method.java:498)
  com.thoughtworks.gauge.execution.MethodExecutor.execute(MethodExecutor.java:32)
  com.thoughtworks.gauge.execution.HooksExecutor$TaggedHookExecutor.executeHook(HooksExecutor.java:99)
  com.thoughtworks.gauge.execution.HooksExecutor$TaggedHookExecutor.execute(HooksExecutor.java:85)
  com.thoughtworks.gauge.execution.HooksExecutor.execute(HooksExecutor.java:42)
  com.thoughtworks.gauge.processor.MethodExecutionMessageProcessor.executeHooks(MethodExecutionMessageProcessor.java:54)
  com.thoughtworks.gauge.processor.SuiteExecutionEndingProcessor.process(SuiteExecutionEndingProcessor.java:26)
  com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:86)
  com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:37)

Successfully generated html-report to => C:\work\testGaugeTemplates\java\art\reports\html-report
Specifications: 0 executed      0 passed        0 failed        0 skipped
Scenarios:      0 executed      0 passed        0 failed        0 skipped
```

**Comment by pablodcar** [comment]
I'm having the same error with maven and Java.
1. gauge --init java
2. Set a pom.xml with maven plugin
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.5</version>
                <configuration>
                    <source>1.7</source>
                    <target>1.7</target>
                </configuration>
            </plugin>
```

3. mvn clean test

Environment:
gauge -v
```
Gauge version: 0.8.0

Plugins
-------
html-report (3.1.0)
java (0.6.0)
xml-report (0.2.0)
```
go version
```go version go1.7.3 darwin/amd64```
java -version
```
java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b14, mixed mode)
```

**Comment by TShireesha** [comment]
Does this issue got resolved?...
Im getting same error still with the new released version of gauge 0.8.0 and plugin 0.3.1

**Comment by BugDiver** [comment]
@pablodcar @TShireesha Could you try updating `gauge-java` dependency in your `pom.xml`.
The latest released gauge version is `0.6.1`.

**Comment by TShireesha** [comment]
Thanks a lot...It worked.

**Comment by naveen7036** [comment]
Today I started exploring Gauge, and I have taken all latest plugins.
Installed Gauge, java, maven in Intellej  and tried running the code present in below link.
https://github.com/getgauge/gauge/wiki/Getting-Started-in-3-Steps

Feeling verymuch rough to understand the issue here. Can someone help me here?

I am getting the below errors:

Testing started at 8:03 PM ...
D:\Work\Automation\Gauge\gauge.exe run --machine-readable --hide-suggestion --simple-console D:/Work/Automation/Gauge/GSearchDemo/specs
{"type":"out","message":"com.google.protobuf.InvalidProtocolBufferException: Message missing required fields: messageType"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.google.protobuf.UninitializedMessageException.asInvalidProtocolBufferException(UninitializedMessageException.java:81)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.google.protobuf.AbstractParser.checkMessageInitialized(AbstractParser.java:71)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:168)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:180)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:185)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:49)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat gauge.messages.Messages$Message.parseFrom(Messages.java:21442)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:67)"}
{"type":"out","message":"\r"}
{"type":"out","message":""}
{"type":"out","message":"\tat com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:37)\r"}
{"type":"out","message":"com.google.protobuf.InvalidProtocolBufferException: Message missing required fields: messageType\r"}
{"type":"out","message":""}
Failed: Before Suite
Message: Connection closed [127.0.0.1:51508] cause: read tcp 127.0.0.1:51503->127.0.0.1:51508: wsarecv: An existing connection was forcibly closed by the remote host.


Failed: After Suite
Message: write tcp 127.0.0.1:51503->127.0.0.1:51508: use of closed network connection

{"type":"out","message":"Successfully generated html-report to =\u003e D:\\Work\\Automation\\Gauge\\GSearchDemo\\reports\\html-report\\index.html"}
{"type":"out","message":""}
{"type": "out", "message": "Specifications:	0 executed	0 passed	0 failed	0 skipped"}
{"type": "out", "message": "Scenarios:	0 executed	0 passed	0 failed	0 skipped"}
{"type": "out", "message": ""}
{"type": "out", "message": "Total time taken: 133ms"}
{"type":"notification","notification":{"title":"Gauge Telemetry","message":"This installation of Gauge collects usage data in order to help us improve your experience.\n\u003ca href=\"https://gauge.org/telemetry\"\u003eRead more here\u003c/a\u003e about Gauge telemetry.","type":"info"}}

Process finished with exit code 1

**Comment by Apoorva-GA** [comment]
Hi could you share your project it will be easier to replicate the issue.
Also please check if the versions of dependencies and plugin in your `pom.xml` file are latest.

**Comment by akalaiselvan** [comment]
Iam getting this error while running my spec. i was working fine ... i did few changes in folder structure , after that not even a single step is running ..pls help me out 

"C:/Program Files/Gauge/bin/gauge.exe" run --machine-readable --hide-suggestion --simple-console D:/QTPCO/AUTOIT/specs/test.spec:40
Testing started at 5:57 PM ...
com.google.protobuf.InvalidProtocolBufferException: Message missing required fields: stepValidateRequest.numberOfParameters


	at com.google.protobuf.UninitializedMessageException.asInvalidProtocolBufferException(UninitializedMessageException.java:81)


	at com.google.protobuf.AbstractParser.checkMessageInitialized(AbstractParser.java:72)

	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:196)

	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:209)


	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:214)


	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:49)


	at gauge.messages.Messages$Message.parseFrom(Messages.java:21451)

	at com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:67)


	at com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:37)


com.google.protobuf.InvalidProtocolBufferException: Message missing required fields: stepValidateRequest.numberOfParameters


[ValidationError] D:QTPCOAUTOITspecs	est.spec:41 Connection closed [127.0.0.1:56307] cause: read tcp 127.0.0.1:56303->127.0.0.1:56307: wsarecv: An existing connection was forcibly closed by the remote host. => 'it prints'Successfully generated html-report to => D:\QTPCO\AUTOIT\reports\html-report\index.html

Specifications:	0 executed	0 passed	0 failed	0 skipped
Scenarios:	0 executed	0 passed	0 failed	0 skipped

Total time taken: 42ms

 
Process finished with exit code 0
